Split book pdfs into chapter pdfs

markmlearneq opened the pull request:
Goalkicker channel link:
https://api.studio.learningequality.org/channels/b17368a2c9925bb58391172f2d336221/edit

Screenshots of Goalkicker books split by chapters:

<img width="1439" alt="kolibri-studio-goalkicker-split-chapters-2019-01-25" src="https://user-images.githubusercontent.com/45840473/51774762-4b36fc00-20a8-11e9-8b5d-f40e451f3c14.png">
<img width="1438" alt="kolibri-studio-goalkicker-chapter-preview-2019-01-25" src="https://user-images.githubusercontent.com/45840473/51774782-568a2780-20a8-11e9-82b0-7242467c5943.png">

Commit:
Split book pdfs into chapter pdfs

ivanistheone commented on sushichef.py:
We have that info in the `title` already so no need to repear.

A better alternative would be to set the description to something like `Chapter n of the Goalkicker book on book_info['subject']` which contains the parent book title, this way is a teacher uses just one chapter of the book in another context, readers will still know where to get the other chapeters.

ivanistheone commented on sushichef.py:
The `chapter['title']` strings seems to contain ligatures (because generated by fancy LaTeX fonts), see https://github.com/learningequality/sushi-chef-goalkicker-tech-books/issues/4 

I recommend you "clean up" the title using something like
```
chapter_title = chapter['title']
for ligature, equiv in LIGATURES.items():
    chapter_title = chapter_title.replace(ligature, equiv)
```
then use `chapter_title` instead of `chapter['title']` everywhere in code below.

Use the `LIGATURES` lookup table from here https://github.com/learningequality/sushi-chef-goalkicker-tech-books/issues/4

ivanistheone replied:
Also, while you're doing the title-cleanup, might as well add get rid of the `®` and `©` symbols because they are a bit annoying. We know `Java` is `®` no need to be reminded every time the word appears!

You could do this by adding two rows to the `LIGATURES` lookup table:
```
  '®': '',
  '©': '',
```

ivanistheone reviewed (changes requested):
See two small fixes in comments

 - cleanup title
 - better description including book title

and a third thing, 

  -  see if you can come up with a channel thumbnail of some sort https://github.com/learningequality/sushi-chef-goalkicker-tech-books/issues/5

Commit:
Put pdfs for each downloaded book into own directory

Fixes issue with the "Content list" and "About" pdfs being locally
overwritten for each book, since each book outputs "0-Content-list.pdf" and
"1-About.pdf" for the first two pdf chunks, which then resulted in every book
resource on Kolibri Studio showing the "Content list" and "About" pdfs for the
last alphabetical book (Xamarin.Forms)

Commit:
Use more descriptive chapter descriptions

Commit:
Replace ligatures with non-ligature equivalents

ivanistheone commented:
> markmlearneq force-pushed ... markmlearneq:pdf-splitting 

Nice! Congrats on learning about `git rebase` !

<img width="477" alt="screen shot 2019-02-18 at 9 01 57 am" src="https://user-images.githubusercontent.com/163966/52955898-df207e80-335b-11e9-87c8-f1d8517adfee.png">

ivanistheone commented on sushichef.py:
@markmlearneq Minor nitpick, but worth mentioning that this should be:

            download_dir = os.path.join('downloads', 'book_' + str(book_counter).rjust(2, '0') + '--' + book_info['subject'])

to make sure it works for Windows and Mac/Linux path separators.

markmlearneq replied:
Ah, yes, thanks for pointing that out
